internal/main.go

```diff
@@ -114,6 +114,25 @@ func main() {
 	}
 }
 
+func getMetadataProvider(providerName string) (func() (providers.Metadata, error), error) {
+	switch providerName {
+	case "azure":
+		return azure.FetchMetadata, nil
+	case "digitalocean":
+		return digitalocean.FetchMetadata, nil
+	case "ec2":
+		return ec2.FetchMetadata, nil
+	case "gce":
+		return gce.FetchMetadata, nil
+	case "packet":
+		return packet.FetchMetadata, nil
+	case "openstack-metadata":
+		return openstackMetadata.FetchMetadata, nil
+	default:
+		return nil, errors.New("unknown provider")
```

Comment:
This is a super tiny nit, but I prefer the pattern of doing a `var ErrUnknownProvider = ...` at the top of the file and returning it here. Makes it easier to expose known error values for tests and such.